src/ctype/CTypeNested.spec.ts

```diff
@@ -0,0 +1,124 @@
+import CType from './CType'
+import Identity from '../identity/Identity'
+import ICType from '../types/CType'
+import Claim from '../claim/Claim'
+// import CTypeUtils from './CTypeUtils'
+
+describe('CType', () => {
+  const identityAlice = Identity.buildFromURI('//Alice')
+
+  const passportCType: ICType['schema'] = {
+    $id: 'http://example.com/ctype-1',
+    $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+    title: 'test',
+    properties: {
+      fullName: { type: 'string' },
+      passportIdentifer: { type: 'string' },
+      streetAddress: { type: 'string' },
+      city: { type: 'string' },
+      state: { type: 'string' },
+    },
+    type: 'object',
+  }
+
+  const kycCType: ICType['schema'] = {
+    $id: 'http://example.com/ctype-1',
+    $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+    title: 'test',
+    properties: {
+      ID: { type: 'string' },
+      number: { type: 'string' },
+      name: { type: 'string' },
+    },
+    type: 'object',
+  }
+
+  const fromPassportCType: ICType = {
+    schema: passportCType,
+    owner: identityAlice.address,
+    hash: '',
+  }
+
+  const fromKYCCType: ICType = {
+    schema: kycCType,
+    owner: identityAlice.address,
+    hash: '',
+  }
+
+  const passport = CType.fromCType(fromPassportCType)
+
+  const kyc = CType.fromCType(fromKYCCType)
+
+  const claimContents = {
+    fullName: 'Archer Macdonald',
+    passportIdentifer: '34jd83jd',
+    streetAddress: '111 reichenberger Strasse',
+    city: 'Berlin',
+    state: 'Germany',
+    ID: '82740927593982508378-0294584...',
+    number: '345678',
+    name: 'Archer Macdonald',
+  }
+
+  it('verifies the nested structure', () => {
+    const nested: ICType['schema'] = {
+      $id: 'http://example.com/ctype-1',
+      $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+      title: 'test',
+      type: 'object',
+      properties: {
+        fullName: {
+          $ref: `${passport.schema.$id}#/properties/fullName`,
+        },
+        passportIdentifer: {
+          $ref: `${passport.schema.$id}#/properties/passportIdentifer`,
+        },
+        streetAddress: {
+          $ref: `${passport.schema.$id}#/properties/streetAddress`,
+        },
+        city: {
+          $ref: `${passport.schema.$id}#/properties/city`,
+        },
+        state: {
+          $ref: `${passport.schema.$id}#/properties/state`,
+        },
+        ID: {
+          $ref: `${kyc.schema.$id}#/properties/ID`,
+        },
+        number: {
+          $ref: `${kyc.schema.$id}#/properties/number`,
+        },
+        name: {
+          $ref: `${kyc.schema.$id}#/properties/name`,
+        },
+      },
+    }
```

Comment:
As mentioned within the pull request, I am not able to valid deeply nested `CType`s. Need some help with this, plus I think I need a function that will take the required `properties` of a given `CType` and spread them as shown within the `nested` object.